### PR TITLE
Id 3530 add binary cache to build dependencies

### PR DESCRIPTION
**Pull request by equinor-ruaj**
Build time down to ~10-20 minutes

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -3,18 +3,27 @@ FROM debian:bullseye as cppbuild
...
+
+
 
 FROM cppbuild as generate_package
+ARG FEED_ACCESSTOKEN
+ARG FEED_URL
+RUN curl -L https://raw.githubusercontent.com/Microsoft/artifacts-credprovider/master/helpers/installcredprovider.sh  | sh
+ENV NUGET_CREDENTIALPROVIDER_SESSIONTOKENCACHE_ENABLED true
+RUN mono /usr/local/bin/nuget.exe sources add -name "ioc-cpp-packages" -Source ${FEED_URL} -Username "docker" -Password ${FEED_ACCESSTOKEN}
+ENV VCPKG_BINARY_SOURCES 'clear;nuget,ioc-cpp-packages,readwrite'
```
Comment (erikmaland) [inline Dockerfile:26]: Does this logic belong in the Dockerfile? This means that the cache cannot be built locally without providing the nuget feed url and access token, doesn't it?

**Review by erikmaland — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (equinor-ruaj) [reply Dockerfile:26]: From what I understand the feed_url and access token must be passed to the Dockerfile in order to use a private nuget feed as we want here...
Alternatives could be to add (and maintain...) a separate Dockerfile-local
Or add some conditional logic like `RUN if [ "access token and feed url is set ]; then <add feed etc.> ; else continue without feed ; fi`

Others?

Comment (erikmaland) [reply Dockerfile]: I do like the solution of fetching from the nuget feed if possible, and continue without it if not. But what will happen in the current implementation if the image is built without providing these arguments?

Comment (equinor-ruaj) [reply Dockerfile]: Now builds locally with and without setting `FEED_ACCESSTOKEN`

**Review by erikmaland — APPROVED** [review]

**Pull request merged**